packages/@coorpacademy-components/src/atom/icon/index.js

```diff
@@ -0,0 +1,99 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
+import {fas} from '@fortawesome/pro-solid-svg-icons';
+import {library} from '@fortawesome/fontawesome-svg-core';
+import toLower from 'lodash/fp/toLower';
+import merge from 'lodash/fp/merge';
+import getOr from 'lodash/fp/getOr';
+import Color from 'colorjs.io';
+import style from './style.css';
+
+// FontAwesome documentation for where to find icon names: https://docs.fontawesome.com/web/add-icons/svg-icon-names
+// Search for FontAwesome icons: https://fontawesome.com/search
+
+library.add(fas);
+
+const DEFAULT_PRESET = 'm';
+const DEFAULT_WRAPPER_SIZE = 40;
+const DEFAULT_COLOR = 'hsl(0, 0%, 32%)';
+
+const SIZE_CONFIGS = {
+  s: {
+    faSize: 'sm',
+    wrapperSize: 32
+  },
+  m: {
+    faSize: 'lg',
+    wrapperSize: DEFAULT_WRAPPER_SIZE
+  },
+  xl: {
+    faSize: 'xl',
+    wrapperSize: 48
+  }
+};
+
+const getIconColor = (backgroundColor, defaultColor = DEFAULT_COLOR) => {
+  if (!backgroundColor) return defaultColor;
+  try {
+    const color = new Color(backgroundColor);
+    color.hsl.l = 32;
+    return color.toString();
```

Comment:
tu peux chainer le tout.
ex: `new Color("slategray").to("hsl").set("h", 12).to('srgb').toString({format: 'hex'});`
![Capture d’écran 2024-04-03 à 14 49 44](https://github.com/CoorpAcademy/components/assets/2601132/7c38e580-1a02-44e6-834c-1f7f781c49e7)

 du coup
```suggestion
    return new Color(backgroundColor).to('hsl').set({l: 32}).toString('hex');
```
(marche aussi avec `.set('l', 32)`)